Sources/Kommunicate/Classes/KMConversationViewController.swift

```diff
@@ -441,6 +441,37 @@ open class KMConversationViewController: ALKConversationViewController {
                 return
             }
             weakSelf.isClosedConversationViewHidden = true
+            guard let zendeskAcckountKey = ALApplozicSettings.getZendeskSdkAccountKey(),
+                  !zendeskAcckountKey.isEmpty else { return }
+            // if zendesk is integrated, create a new conversation instead of restarting the conversation
+            let zendeskHandler = KMZendeskChatHandler.shared
+            zendeskHandler.resetConfiguration()
+            zendeskHandler.initiateZendesk(key: zendeskAcckountKey)
+            weakSelf.loadingStarted()
+            // Create a new conversation 
+            let kmConversation = KMConversationBuilder()
+                          .useLastConversation(false)
+                          .build()
+            Kommunicate.createConversation(conversation: kmConversation) { result in
+              switch result {
+               case .success(let conversationId):
+                  ALApplozicSettings.setLastZendeskConversationId(NSNumber(value: Int(conversationId) ?? 0))
+                  let convViewModel = ALKConversationViewModel(contactId: nil, channelKey: NSNumber(value: Int(conversationId) ?? 0), localizedStringFileName: Kommunicate.defaultConfiguration.localizedStringFileName, prefilledMessage: nil)
+                  // Update the viewmodel
+                  weakSelf.viewModel = convViewModel
+                  weakSelf.unsubscribingChannel()
+                  weakSelf.viewModel.contactId = nil
+                  weakSelf.viewModel.prefilledMessage = nil
+                  weakSelf.viewModel.channelKey = NSNumber(value: Int(conversationId) ?? 0)
+                  weakSelf.viewModel.conversationProxy = nil
+                  weakSelf.viewModel.delegate = self
```

Comment:
See if we can move these setup calls to a separate function

Reply:
done